### ISSUE TITLE
Errors on H170 Gaming 3

### ISSUE THREAD

**Issue by PenguinPaul** [xdarklight/gigabyte_ambientled_ctrl#2]
```
paul@DeusVult:~/Scripts/gigabyte_ambientled_ctrl-master$ sudo ./audio_led_mode.sh BeatMode
./audio_led_mode.sh: line 36: echo: write error: Invalid argument
./audio_led_mode.sh: line 41: echo: write error: Invalid argument
Enabling beat/tempo mode for audio LED...
./audio_led_mode.sh: line 46: /sys/class/gpio/it87_gp82/direction: No such file or directory
./audio_led_mode.sh: line 47: /sys/class/gpio/it87_gp82/value: No such file or directory
./audio_led_mode.sh: line 49: /sys/class/gpio/it87_gp83/direction: No such file or directory
./audio_led_mode.sh: line 50: /sys/class/gpio/it87_gp83/value: No such file or directory
```

**Comment by xdarklight** [comment]
could you please paste the output of ``zgrep CONFIG_GPIO_IT87 /proc/config.gz`` and ``lsmod | grep it87``?

**Comment by PenguinPaul** [comment]
`gzip: /proc/config.gz: No such file or directory`

and `lsmod | grep it87` outputs nothing.

**Comment by xdarklight** [comment]
what Linux distro is this (Ubuntu, Debian, Arch Linux, etc...)?

these scripts have a dependency: they require the gpio_it87 kernel driver to work.
on Arch Linux for example this driver is not enabled by default (you can either re-build the kernel or use an out-of-tree build)

**Comment by PenguinPaul** [comment]
Ubuntu, kernel 4.8.

**Comment by xdarklight** [comment]
it seems that the kernel config is stored somewhere else on Ubuntu.
What does ```grep CONFIG_GPIO_IT87 /boot/config-$(uname -r)``` give you?

(if that still gives you errors: could you please provide a link from where you got the Ubuntu 4.8 kernel)

**Comment by PenguinPaul** [comment]
`paul@DeusVult:/usr/share/themes$ grep CONFIG_GPIO_IT87 /boot/config-$(uname -r)
CONFIG_GPIO_IT87=m`

**Comment by PenguinPaul** [comment]
After running `sudo modprobe gpio_it87` the script now runs.

**Comment by xdarklight** [comment]
OK, does it also turn the LEDs on or off?
if not, could you please provide the output of the following commands:
* ```cat /sys/class/dmi/id/board_name```
* ```grep "" /sys/class/gpio/gpiochip*/{label,base}```
* ```echo 462 > /sys/class/gpio/export && echo out > /sys/class/gpio/it87_gp26/direction && echo 0 > /sys/class/gpio/it87_gp26/value``` (then try the same command with "echo 1 ..." again which should turn on the LEDs)

**Comment by PenguinPaul** [comment]
It is fully functional now.

**Comment by xdarklight** [comment]
Fixed with 40978bec0ed8eb08bc47c455b34a1dbb9a6e8aca then - thanks for reporting back!

**Comment by PenguinPaul** [comment]
For the record, it does support beat(?) mode and does flash with the music when played out of the read motherboard sound outputs (but not from the front case outputs)

**Comment by xdarklight** [comment]
@PenguinPaul corrected with 428c6e52d1e0f1de27fc5568d7dd25a295349f63 - thanks!
